flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/SpillableSubpartitionView.java

```diff
@@ -169,6 +169,13 @@ public void releaseAllResources() throws IOException {
 			if (spilled != null) {
 				spilled.releaseAllResources();
 			}
+			// we are never giving this buffer out in getNextBuffer(), so we need to clean it up
+			synchronized (buffers) {
+				if (nextBuffer != null) {
+					nextBuffer.recycle();
+					nextBuffer = null;
```

Comment:
Why do we need this `nextBuffer` field? Is it because there can be multiple views talking to the same subpartition and buffers are kind of pre fetched for each view (one buffer ahead is pre fetched)?

Reply:
To be honest, I don't know - looking at it further, I think you are right: it only makes sense with multiple views attached to the same subpartition; but we are not using that. Nevertheless, I don't really like to mess with this code as of now and it is not really critical either.

Reply:
hmm ok. However if we do not use anything like that, this field could be dropped and code would simplify. At least something to keep in mind in the future.